src/main/kotlin/cacophony/codegen/BlockLabel.kt

```diff
@@ -1,3 +1,23 @@
 package cacophony.codegen
 
+import cacophony.semantic.syntaxtree.Definition
+
 data class BlockLabel(val name: String)
+
+object FunctionBodyLabel {
+    private val functionToId = mutableMapOf<Definition.FunctionDeclaration, Int>()
+    private var lastId = 0
+
+    operator fun invoke(function: Definition.FunctionDeclaration): BlockLabel {
+        if (!functionToId.containsKey(function)) {
+            functionToId[function] = lastId++
+        }
+
+        return BlockLabel(
+            when (function.identifier) {
+                "<program>" -> "start"
+                else -> "${function.identifier}_${function.arguments.size}_id${functionToId[function]}"
```

Comment:
In my opinion the asm label should not change by simply changing the order of the functions. I thought of something like parentlabel_functionname_argcount when I wrote the comment yesterday, but I just noticed that it may lead to some collisions. 

If you have an idea how to make a label depend only on parent function name, function name & argument counts (or some similar values that do not change based on function order / white spaces etc.) please change it. Otherwise we can leave it as is, and maybe discuss this again in the future once we need it.

Reply:
Any label that is generated using only local data can lead to collisions. Maybe lets leave this until someone has a better idea.

Reply:
That is not true in general. e.g. <parent_label>_[function name length]_[function name]_[argument count] works (but is ugly imo).